control/joystick_commander/src/commander.c

```diff
@@ -56,846 +53,785 @@
-                0.0,
-                MAX_THROTTLE_PEDAL );
-    }
+    const double max_position;
+};
 
 
-    return ret;
-}
 
+// *****************************************************
+// static global data
+// *****************************************************
 
-//
-static int get_steering_setpoint(
-        joystick_device_s * const jstick,
-        double * const angle )
-{
-    int ret = NOERR;
-    int axis_position = 0;
-
+/**
+ * @brief Commander data.
+ *
+ * Static definitions of the variables that store the relevant
+ * information about what the current variables are
+ * 
+ * The validity of the commander pointer is how to determine if
+ * the commander is available for use
+ *
+ */
+static struct commander_data_s commander_data = { 0, 0.0, 0.0, 0.0, 0.0 };
+static struct commander_data_s* commander = NULL;
 
-    // read axis position
-    ret = jstick_get_axis(
-            jstick,
-            JSTICK_AXIS_STEER,
-            &axis_position );
+/**
+ * @brief Setpoint Data
+ *
+ * Static definitions for brake, steering and throttle setpoints
+ *
+ */
+static struct commander_setpoint_s brake_setpoint =
+    { 0.0, JOYSTICK_AXIS_BRAKE, 0.0, MAX_BRAKE_PEDAL };
 
+static struct commander_setpoint_s throttle_setpoint =
+    { 0.0, JOYSTICK_AXIS_THROTTLE, 0.0, MAX_THROTTLE_PEDAL };
 
-    // if succeeded
-    if( ret == NOERR )
-    {
-        // set steering wheel angle set point - scale to max:min
-        // note that this is inverting the sign of the joystick axis
-        (*angle) = jstick_normalize_axis_position(
-                axis_position,
-                MIN_STEERING_WHEEL_ANGLE,
-                MAX_STEERING_WHEEL_ANGLE );
-    }
+static struct commander_setpoint_s steering_setpoint =
+    { 0.0, JOYSTICK_AXIS_STEER, MIN_STEERING_WHEEL_ANGLE, MAX_STEERING_WHEEL_ANGLE };
 
 
-    return ret;
-}
+// *****************************************************
+// static definitions
+// *****************************************************
 
 
+// *****************************************************
+// Function:    get_setpoint
+// 
+// Purpose:     Retrieve the data from the joystick based on which axis is
+//              selected and normalize that value along the scale that is
+//              provided.
+// 
+// Returns:     int - ERROR or NOERR
+// 
+// Parameters:  setpoint - the setpoint structure containing the range limits
+//                         the joystick axis, and the storage location for the
+//                         requested value
 //
-static int get_disable_button(
-        joystick_device_s * const jstick,
-        unsigned int * const state )
+// *****************************************************
+static int get_setpoint( struct commander_setpoint_s* setpoint )
 {
-    int ret = NOERR;
-    unsigned int btn_state = JOYSTICK_BUTTON_STATE_NOT_PRESSED;
-
+    int return_code = ERROR;
 
-    // default state is disabled/not-pressed
-    (*state) = 0;
+    if ( setpoint != NULL )
```

Comment:
The return_code is initialized to ERROR so it should return ERROR by default unless everything works correctly.